pkg/transform/kubernetes/populators.go

```diff
@@ -161,3 +161,34 @@ func populateContainers(containers []spec.Container, cms []spec.ConfigMapMod, se
 	log.Debugf("containers after populating health: %s", string(b))
 	return cnts, nil
 }
+
+// Since we are automatically creating pvc from
+// root level persistent volume and entry in the container
+// volume mount, we also need to update the pod's volume field
+func populateVolumes(containers []api_v1.Container, volumeClaims []spec.VolumeClaim,
```

Comment:
in some places it is unavoidable, but wherever we can I wanna avoid passing the entire struct. Also makes it resusable, like in here https://github.com/kedgeproject/kedge/pull/214/files#diff-e6ad971f18906f2176a088cf3bb5b070R54